src/index.d.ts

```diff
@@ -168,6 +198,11 @@ export declare class EmptyObservableError extends Error {
   constructor()
 }
 
+type StateObservableInput<T, E> =
+  | EffectObservable<T, E>
+  | StateObservable<T, E>
+  | DefaultedStateObservable<T, E>
+
```

Comment:
This was needed because calling `state(someOtherState$)` should be a noop on types, but for some reason it wasn't able to pick up any of the generics - The result would be a `StateObservable<unknown, unknown>`.